API/common/service_impl/learn_knowledge_index.py

```diff
@@ -70,25 +74,62 @@ async def _retrieve_chunks_from_learn_knowledge_index_service(vector_values, fil
     return convert_chunks_to_json(result.json())
 
 
-async def retrieve_chunks_for_atomic_task(task_info):
-    vector_values = await _embedding_text_to_vector(task_info)
+def trim_command_and_chunk_with_invalid_params(command, chunk):
+    """
+    Trim the incorrect parameters in command and suggest the possible correct parameters in returned chunk.
+    Args:
+        command: the command contains incorrect parameters
+        chunk: a chunk contains all parameters related to the command
+    Returns: A tuple. The first element is the command with only correct parameter.
+    The second element is the chunk with possible correct parameters
+    """
+    chunk_copy = chunk.copy()
+    cmd_sig, params = parse_command_info(command)
+    cmd_params = []
+    unmatched_params = []
+    for param in params:
+        if _pop_param_in_chunk(param, chunk_copy):
+            cmd_params.append(param)
+        else:
+            unmatched_params.append(param)
+    chunk_copy['optional parameters'] = []
+    for param in unmatched_params:
+        similar_params = _find_top_n_similar_params(param, chunk['optional parameters'])
+        chunk_copy['optional parameters'].extend(similar_params)
+    chunk_copy['optional parameters'] = _dedup_param_group(chunk_copy['optional parameters'])
+    valid_cmd = cmd_sig + ' ' + ' '.join(cmd_params)
+    return valid_cmd, chunk_copy
 
-    is_command = task_info.startswith("az ")
-    if is_command:
-        command_info, parameters_info = parse_command_info(task_info)
-        chunk_items = await _retrieve_chunks_from_learn_knowledge_index_service(vector_values, filter_command=command_info)
-        # If the split subtask is a correct command, then accurately search for chunks related to this command
-        if chunk_items:
-            return merge_chunks_by_command(chunk_items)[0]
 
-        else:
-            # If it is a command that cannot be accurately matched, it indicates that this command is fabricated or expired. Then, use vector query for similarity search
-            chunk_items = await _retrieve_chunks_from_learn_knowledge_index_service(vector_values)
-            return merge_chunks_by_command(chunk_items)[0]
+async def retrieve_chunk_for_command(command: str):
+    """
+    Retrieve chunks according to command signature
+    Args:
+        command: full command
+    Returns: a merged chunk that is related to the command
+    """
+    sig, _ = parse_command_info(command)
+    vector_values = await _embedding_text_to_vector(command)
+
+    chunk_items = await _retrieve_chunks_from_learn_knowledge_index_service(vector_values, filter_command=sig)
+    if not chunk_items:
+        chunk_items = await _retrieve_chunks_from_learn_knowledge_index_service(vector_values)
+    chunks = merge_chunks_by_command(chunk_items)
+    return chunks[0] if chunks else None
```

Comment:
When this command cannot be accurately filtered, vector queries may find multiple commands. Is the first command necessarily the most relevant?

Reply:
The usage of `retrieve_chunk_for_command` has been changed during my development. I think we shouldn't return any chunk if there is no chunk accurately matched. I will remove this `if` condition.